### PR TITLE
feat: app builder backend

### PR DESCRIPTION
**Pull request by joshbalfour**
**Detail as per issue below (required):**
A bit big as it's adding a whole new service.
Dockerfile is top-level as it's reusable across all packages and needs the context of the whole monorepo.

Thinking that we could later extract the cdk components used into a top level package which could be reused across the other packages, which cdk suits itself to quite nicely.

Fixes: #4523, #4604

### CODE REVIEW

File: packages/app-builder-backend/src/entities/Author.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { ObjectType, Field, ID } from 'type-graphql'
```
Comment (willmcvay) [inline packages/app-builder-backend/src/entities/Author.ts:1]: Tiny thing but kebab-case for files - it's not a big deal as it's only stub data so don't feel you have to change it

File: packages/app-builder-backend/package.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  "main": "./src/index.ts",
+  "scripts": {
+    "start": "ts-node .",
+    "test": "echo '...skipping...'",
+    "build": "tsc",
+    "lint": "eslint --cache --ext=ts,tsx,js src --fix",
+    "release": "cdk deploy --require-approval never",
+    "publish": "yarn npm publish --access public",
+    "conf": "echo '...skipping...'",
+    "synth": "cdk synth",
```
Comment (willmcvay) [inline packages/app-builder-backend/package.json:20]: What does this do? Just interested :-)

File: packages/app-builder-backend/package.json [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    "type": "git",
+    "url": "git+git@github.com:reapit/foundations.git"
+  },
+  "license": "MIT",
+  "author": "Reapit Ltd",
+  "main": "./src/index.ts",
+  "scripts": {
+    "start": "ts-node .",
+    "test": "echo '...skipping...'",
+    "build": "tsc",
```
Comment (willmcvay) [inline packages/app-builder-backend/package.json:15]: Does this include your node modules correctly? Great if we can get away with no webpack

**Review by willmcvay — APPROVED** [review]
LGTM 👍🏻

File: packages/app-builder-backend/package.json (hunk 3 above)
Comment (joshbalfour) [reply packages/app-builder-backend/package.json:15]: yep! the node modules get bundled into the docker container on build
https://github.com/reapit/foundations/blob/feat/%234523-app-builder-backend/Dockerfile#L50

File: packages/app-builder-backend/package.json (hunk 2 above)
Comment (joshbalfour) [reply packages/app-builder-backend/package.json:20]: it generates the cloudformation template from the cdk code and validates it, useful if you make a cdk code change and want to make sure that it'll deploy without having to actually deploy it

**Pull request merged**